test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -2017,4 +2028,39 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthFo
       });
     });
   });
-});
+  
+  Expect.describe('Registration Flow', function () {
+    itp('does not show the registration button if settings.registration.click is not set', function () {
+      return setup().then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration is empty object', function () {
+      var registration =  {};
+      return setupRegistrationButton(registration.click, registration.label, registration.text).then(function (test) {
```

Comment:
It's a bit strange to me that we're creating an object, but passing in undefined's (since nothing is set), and then resetting these properties on a new object in the setupRegButton function.

It would be clearer to me to just pass in the registration object to setupRegButton:
```javascript
function setupRegistration(registration) {
  return setup({ registration: registration });
}

var registration = {};
return setupRegistration(registration).then(/* stuff */);
```